### DISCUSSION TITLE
"YANKED" doesn't seem ideal (informal dev-speak-ish)

### DISCUSSION THREAD

**Discussion by jblaine** [olivierlacan/keep-a-changelog#470] [General]
Nitpicky, but I think RETRACTED or WITHDRAWN is more appropriate for an effort to make a common changelog format.

**Comment by shadowspawn** [comment]
This was discussed at length in #195, skip to the final comment to see sticking with YANKED but change it if you want.

**Comment by Bananeweizen** [comment]
"Yanked" just doesn't fit to the very first goal of the project, "being for humans". That term is not understood by a huge amount of non-native speakers, as a short survey around colleagues shows me. That problem should have much more weight than whether an arbitrary programming language uses that term with the same meaning as intended by this project.

To me it is as if you want to teach people how to eat healthy, but then chose to use terminology that is only understood by doctors and nutritionists.

**Comment by pboling** [comment]
A changelog is for technical people.  Technology, and software in particular, have lexicons that won't always be "basic" English.  Technologists should learn what "YANKED" means, and this will help them in improving skills.
"YANKED" is very specific, and not equivalent to "WITHDRAWN" or "RETRACTED" as discussed at length in #195.  The increase in ambiguity, decrease in specificity, for native speakers, or people familiar with the acute terminology, by using a less precise term, is not worth it.

**Comment by jblaine** [comment]
I can't really say it better than @Bananeweizen did or others in #195 did. And #195 has no solid (IMO) argument against "WITHDRAWN". It means exactly the same thing as "YANK" and isn't developer slang. The idea that "yank" is to be the "new truth" to mean withdrawn in a _global changelog format_, because a few software development language tools use that word, doesn't hold any water to me. What "npm and gem do" isn't the gold standard for "right for all cases because some people got used to that word". There are 10 other modern and heavily used software development language ecosystems that have no "yank" defined.

There's zero ambiguity about the following and it is more intuitively natural worldwide by anyone. I highly suspect this "YANKED" topic will keep coming up, and there's a reason it will.

## [1.0.0] - 2017-06-20
### Added
- User profiles

### Changed
- Polish terminology corrections

## [0.9.8] - 2017-06-03 [WITHDRAWN]

## [0.9.0] - 2017-05-10
### Added
- Topic navigation.

**Comment by SMUnlimited** [comment]
I can't see commerical companies using "yanked" as it would appear unprofessional if clients got hold of it. 

If the end result of the original thread was to "replace with a term of your choice" that is more fitting to your environment, perhaps that should be the only amendment in the FAQ.

**Comment by LucidOne** [comment]
:+1: for adding WITHDRAWN as an official alternative to YANKED to the FAQ.

**Comment by pboling** [comment]
[Yanked is used in the press](https://www.cnet.com/news/yanked-ios-app-clueful-is-back-as-free-web-software/), so it isn't developer slang.  I showed many other instances of press use in #195.  

Withdrawn is used in the press also, but it seems to me that it is usually in less technical contexts for less technical audiences.

For a technical audience the meaning is not identical, as discussed at length in #195.

If a language ecosystem has no way to `yank` software globally, then it makes sense that there is no `yank` defined!  As such those language ecosystems have no bearing on the proper term for this concept.  Rubygems, Rust, and the "packagecloud.io" service all seem to be in agreement.

https://packagecloud.io/docs defines `push` and `yank`.

I haven't seen any documented evidence of `withdrawn` being used to unambiguously indicate a *global removal of a software package from a repository of software packages*.

If you want to use `withdrawn` as an indication that the release is unsupported, but still might be available for download, then that is literally a **different meaning**.

**Comment by LucidOne** [comment]
I don't really care that much, it's not like anyone who reads CHANGELOGs is really going to care. I just think YANKED sounds very informal and I am not alone. My personal preference would be RETRACTED or  the historical term NUKED which has been used prior to Ruby, Rust, etc. It is clearly past time for arguing this though.

A compromise would be to use YANKED when the programming language's package management system  uses YANKED and use WITHDRAWN when a release is made no longer avail by human intervention.

Anyways, my reading of #195 was that the conclusion was "Use what ever you want", which isn't much of a standard. I'm just suggesting it would be better to pick one or two identifiers and move on.
